controller.py

```diff
@@ -638,6 +645,37 @@ def remove_items(self, collection_details):
 
         return feed_response(removal_feed)
 
+    def _in_catalog_subset(self, collection, identifiers_by_urn):
+        """Helper method to find a subset of identifiers that
+        are already in a catalog.
+
+        :param collection: The collection whose catalog we're checking.
+
+        :param identifiers_by_urn: A dictionary mapping URNs to identifiers,
+        like the one returned by Identifier.parse_urns.
+
+        :return: a 2-tuple (in_catalog_ids,
+        match_clause). `in_catalog_ids` is a set of Identifier IDs
+        representing the subset of identifiers currently in the
+        catalog. `match_clause` is the SQLAlchemy clause that was used
+        to look up the matching subset.
+        """
+        # Extract the identifier IDs from the dictionary.
+        identifier_ids = [x.id for x in identifiers_by_urn.values()]
+
+        # Find the subset of identifiers that are in the catalog, so
+        # we know to give them a 200 message after deletion.
+        table = collections_identifiers.c
+        identifier_match_clause = and_(
+            table.identifier_id.in_(identifier_ids),
+            table.collection_id == collection.id
```

Comment:
The absence of this line was what caused the bug in my earlier branch.

Reply:
👌